### DISCUSSION TITLE
FromSheet bug report

### DISCUSSION THREAD

**Discussion by SpyBott** [spatie/simple-excel#114] [Q&A]
Hi, 

I try to use simple-excel to read an excel contains two spreadsheets. 
The spreadseet, i want to use, is the second. Your documentation say that it's with the function fromSheet(int).
But when i try this, it read a part of the first spreadsheet.
The first spreadsheet contains graphics and it's maybe that which do the bug ?!  

I try too to use fromSheet with the name of the spreasheet, but it's not work because the param of from sheet must be integer.
Couldn't we set up fromSheet by passing the name of the spreadsheet we want to use?

**Comment by SpyBott** [comment]
More information : 
In SimpleExcelReader, i've doing a var_dump on keys iterator, and i've got : 
int(1)
int(2)
int(1)
int(2)

The iterator has weird values.

**Comment by SpyBott** [comment]
More information : 
I've echo all of Spreadsheet names in the foreach ($this->reader->getSheetIterator() as $key => $sheet) { in SImpleExcelReader->getSheet()
The response is : 
string(20) "???Feuille_1"
string(27) "???data_Feuille_1_1"
string(27) "???data_Feuille_1_2"
string(20) "???Feuille_2"
And i don't understand why i've data_Feuille_1_1 and data_Feuille_1_2 in the var_dump of iterator's keys.

**Comment by SpyBott** [comment]
I modified your SimpleExcelReader class to be able to do a fromSheetName and therefore read a spreadsheet by its name and no longer its index.
It works pretty well.

Do you want me to send you this modification?

**Comment by freekmurze** [comment]
If the code is not too complicated, I'd accept a PR indeed. Please also include tests in the PR 👍

**Reply by SpyBott** [reply]
Oh no, the code is simple, i've juste add a protected string variable SheetName = "";
The function 

    public function fromSheetName(string $sheetName): SimpleExcelReader
    {
        $this->sheetNumber = -1;
        $this->sheetName = $sheetName;
        $this->headers = null;

        return $this;
    }
    
    And modified the function getsheet like this : 
     protected function getSheet(): SheetInterface
    {
        $this->reader->open($this->path);
        foreach ($this->reader->getSheetIterator() as $key => $sheet) {
            if (($this->sheetNumber !== -1 && $key === $this->sheetNumber) || ($this->sheetName != "" && $this->sheetName === $sheet->getName())) {
                break;
            }
        }
        if ($this->sheetNumber !== -1 && $this->sheetNumber !== $key) {
            throw new InvalidArgumentException("Sheet {$key} does not exist in {$this->path}.");
        } else {
            if ($this->sheetName != "" && $this->sheetName === $sheet->getName()) {
                throw new InvalidArgumentException("Sheet {$this->sheetName} does not exist in {$this->path}.");
            }
        }
        return $sheet;
    }

**Comment by SpyBott** [comment]
Pull request created => https://github.com/spatie/simple-excel/pull/117 :)
